scalyr_agent/copying_manager/worker.py

```diff
@@ -354,6 +355,7 @@ def __init__(self, configuration, worker_config_entry, session_id, is_daemon=Fal
 
         # The current pending AddEventsTask.  We will retry the contained AddEventsRequest several times.
         self.__pending_add_events_task = None
+        self.__receive_response_status = {}
```

Comment:
Where is this private field used?

Reply:
https://github.com/scalyr/scalyr-agent-2/blob/d9c9779fb46bb52b4bebd0b02cae37af83c966b6/scalyr_agent/copying_manager/worker.py#L494

Trying to resolve this error (it doesn't):

```
2024-01-26 17:17:51.891Z ERROR [core] [scalyr_agent.scalyr_logging:707] Failed while attempting to scan and transmit logs :stack_trace:
  Traceback (most recent call last):
    File "/Users/anthonyj/_Scalyr/scalyr-agent-2/scalyr_agent/copying_manager/worker.py", line 500, in run
      "Repeatedly failed to parse response due to exception.  Dropping events",
  AttributeError: 'AddEventsTask' object has no attribute '_CopyingManagerWorkerSession__receive_response_status'
```

I defined the attribute in hopes that this was just a sequencing issue in the way the `scalyr_client` works, but this didn't fix it. Happens when we start to hit publishing delays. Easiest way for me to hit it is to Debug Agent;Pause Execution for X minutes;Resume Agent

Since this is experimental, I'm comfortable shipping this bug to the customer so they can provide feedback on the feature.

Reply:
I meant the second one on the line 358. I believe that's a mistake.